gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -0,0 +1,118 @@
+                });
+        TaskProvider<CheckGradleJdkConfigsTask> checkGradleJdkConfigs = rootProject
+                .getTasks()
+                .register("checkGradleJdkConfigs", CheckGradleJdkConfigsTask.class, task -> {
+                    task.getInputGradleDirectory()
+                            .set(generateGradleJdkConfigs
+                                    .get()
+                                    .getOutputGradleDirectory()
+                                    .getLocationOnly()
+                                    .get());
+                    task.getDummyOutputFile()
+                            .set(rootProject.getLayout().getBuildDirectory().file("checkGradleJdkConfigs"));
+                });
+
+        rootProject.getTasks().withType(GradleJdkConfigs.class).configureEach(task -> {
+            task.getDaemonJavaVersion().set(jdksExtension.getDaemonTarget());
+            task.getJavaVersionToJdkDistros()
+                    .putAll(rootProject.provider(() -> JdkDistributionConfigurator.getJavaVersionToJdkDistros(
+                            rootProject, jdkDistributions, jdksExtension)));
+            task.getCaCerts().putAll(jdksExtension.getCaCerts());
+        });
+        TaskProvider<GradleWrapperPatcherTask> wrapperPatcherTask = rootProject
+                .getTasks()
+                .register("wrapperJdkPatcher", GradleWrapperPatcherTask.class, task -> {
+                    task.getOriginalGradlewScript()
+                            .fileProvider(
+                                    rootProject.provider(() -> wrapperTask.get().getScriptFile()));
+                    task.getBuildDir().set(task.getTemporaryDir());
+                    task.getPatchedGradlewScript()
+                            .set(rootProject.file(
+                                    rootProject.getRootDir().toPath().resolve("gradlew")));
+                    task.getGenerate().set(true);
+                    task.dependsOn(generateGradleJdkConfigs);
+                });
+        wrapperTask.configure(task -> {
+            task.finalizedBy(wrapperPatcherTask);
+        });
+
+        rootProject
+                .getTasks()
+                .named(LifecycleBasePlugin.CHECK_TASK_NAME)
+                .configure(check -> check.dependsOn(checkGradleJdkConfigs));
+
+        rootProject.getTasks().register("setupJdks", SetupJdksTask.class, setupJdksTask -> {
+            setupJdksTask.setDescription("Configures the gradle JDK setup.");
+            setupJdksTask.setGroup(GRADLE_JDK_GROUP);
+            setupJdksTask.getProjectDir().set(rootProject.file(rootProject.getPath()));
+            setupJdksTask.getGradlewScript().set(wrapperPatcherTask.get().getPatchedGradlewScript());
+            setupJdksTask.dependsOn(generateGradleJdkConfigs, wrapperPatcherTask);
+        });
+
+        rootProject.getTasks().named("javaToolchains").configure(task -> {
```

Comment:
WIP - hacky solution at the moment to make sure that when running the intelij startup task - https://github.com/palantir/gradle-jdks/pull/360/files#diff-4e8f06bc09ead2db0d11b0b160158e0fc79f6d700c6ae08b22dd523a2504263aR91, we are running both the "checkGradleJdkConfigs" and "javaToolchains".

Ideally I'd like to have a lifecycle task `ideaSetup` that triggers both `checkGradleJdkConfigs` and `javaToolchains`. Atm I have some issues retrieving the available gradle tasks in the Intellij plugin to check if a specific task exists. I wouldn't want to depend on the gradle property `palantir.setup.jdk.setup` as this might break when the flag won't be used anymore/the new workflow will be enabled by default.

Reply:
Do we need this anymore now we stopped calling gradle from intellij?